crates/chain/src/keychain/txout_index.rs

```diff
@@ -445,8 +445,12 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
         keychain: K,
     ) -> super::ChangeSet<K> {
         let descriptor_id = descriptor.descriptor_id();
-        self.keychains_to_descriptors
+        let old_desc = self
+            .keychains_to_descriptors
             .insert(keychain.clone(), (descriptor_id, descriptor.clone()));
+        if let Some((old_desc_id, _)) = old_desc {
+            self.descriptor_ids_to_keychain.remove(&old_desc_id);
+        }
```

Comment:
You're removing the old descriptor id if it got replaced, but you're not removing the old keychain id if the keychain got replaced. Is there a specific reason for this behavior? 

Also, is it really necessary to remove the old descriptor?